siliconcompiler/core.py

```diff
@@ -3852,10 +3854,9 @@ def _setup_node(self, step, index):
         else:
             self.error(f'setup() not found for tool {tool}, task {task}', fatal=True)
 
-        # Need to clear index, otherwise we will skip setting up other indices.
-        # Clear step for good measure.
-        self.set('arg', 'step', None)
-        self.set('arg', 'index', None)
+        # Need to restore step/index, otherwise we will skip setting up other indices.
```

Comment:
This is the preserving of `step/index` args that I was just talking about in our standup.